### PR TITLE
Add Stratis root filesystem blog post

### PR DESCRIPTION
**Pull request by jbaublitz**
Preliminary blog post for Stratis root filesystem feature.

### CODE REVIEW

**Review by mulkieran — CHANGES_REQUESTED** [review]
Have just one request...

File: content/stratis-rootfs.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+for a Linux installation with proper customization of the initramfs, our most
+recent feature provides all of the plumbing to fully support Stratis filesystems
+as the root filesystem of a Linux installation.
+
+### IPC
+Stratis relies on D-Bus for interprocess communication between the daemon and the
+client. D-Bus does not currently ship in the initramfs so the first order of business
+was to choose an alternate form of IPC. Because we require the ability to pass a file
+descriptor from the client to the daemon, this made Unix sockets the only reasonable
+transport mechanism. After evalutating several JSON RPC libraries with Unix socket
```
Comment (mulkieran) [inline content/stratis-rootfs.md:21]: ^evalutating^evaluating

File: content/stratis-rootfs.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+`/dev/stratis/mypool/myfs / xfs defaults,x-systemd.requires=stratis-fstab-setup@[POOL_UUID].service,x-systemd.after=stratis-fstab-setup@[POOL_UUID].service 1 1`
+
+where `POOL_UUID` is replaced by the UUID of the pool containing your filesystems.
+
+This will cause the setup script to be run and the mount will not proceed until it has
+been completed successfully.
+
+### Recovery console
+While we mention above that stratisd could previously be used in the initramfs, there
+is one major caveat: it could started but commands could not be issued. This had
```
Comment (mulkieran) [inline content/stratis-rootfs.md:87]: ^it could...^it could be started but it could not process D-Bus commands

**Review by mulkieran — CHANGES_REQUESTED** [review]
Two requests, otherwise, looks fine to me.

**Review by drckeefe — APPROVED** [review]

**Review by mulkieran — COMMENTED** [review]
This looks fine to me.

File: content/stratis-rootfs.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+works quite well.
+
+### `/etc/fstab` or `.mount` files
+We now also provide a systemd service to manage setting up devices in /etc/fstab.
+For devices that require a passphrase or are critical for a working system, the
+following line can be used:
+
+`/dev/stratis/[STRATIS_SYMLINK] [MOUNT_POINT] xfs defaults,x-systemd.requires=stratis-fstab-setup@[POOL_UUID],x-systemd.after=stratis-fstab-setup@[POOL_UUID]`
+
+The absense of `nofail` here is due to the fact that `nofail` causes the boot to
```
Comment (bgurney-rh) [inline content/stratis-rootfs.md:77]: Spelling correction: "The absence of..."

**Review by bgurney-rh — APPROVED** [review]

**Review by mulkieran — CHANGES_REQUESTED** [review]
One last request from me...please put in a line indicating that boot from root will be available in stratisd 2.4.0. Similar to the multi-threading PR:
```
Remarks
=======
Preliminary multi-threading support will be included in the next `stratisd`
release, 2.4.0.
```

After that, it should be ready to go.

**Review by bgurney-rh — APPROVED** [review]

**Review by mulkieran — CHANGES_REQUESTED** [review]
This looks ready to me. Please close this and open a new PR for final review.

**Pull request closed without merging**